# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/quota/OriginOperations.h
# Commit: f799f35b1c84
# Full Hash: f799f35b1c84a8c16f1cca5e09cda13372dc8528
# Author: Jan Varga <jvarga@mozilla.com>
# Date: 2024-10-10 09:22:33
# Regressor Bug: 1866402
# File Overlap Count: 1
# Description:
#   Bug 1866402 - Change all origin operations which uninit origins to return an array of origin metadata; r=dom-storage-reviewers,jari
#   
#   This is a preparation for unsetting some flags when origins are either cleared
#   or reset.
#   
# ==============================================================================

diff -r 0afeb47f38b0 -r f799f35b1c84 dom/quota/OriginOperations.h
--- a/dom/quota/OriginOperations.h	Wed Oct 09 22:16:49 2024 +0000
+++ b/dom/quota/OriginOperations.h	Wed Oct 09 22:16:50 2024 +0000
@@ -113,7 +113,7 @@
 RefPtr<ResolvableNormalOriginOp<bool>> CreateClearStorageOp(
     MovingNotNull<RefPtr<QuotaManager>> aQuotaManager);
 
-RefPtr<ResolvableNormalOriginOp<bool>> CreateClearOriginOp(
+RefPtr<ResolvableNormalOriginOp<OriginMetadataArray, true>> CreateClearOriginOp(
     MovingNotNull<RefPtr<QuotaManager>> aQuotaManager,
     const Maybe<PersistenceType>& aPersistenceType,
     const mozilla::ipc::PrincipalInfo& aPrincipalInfo);
@@ -124,19 +124,20 @@
     const mozilla::ipc::PrincipalInfo& aPrincipalInfo,
     Client::Type aClientType);
 
-RefPtr<ResolvableNormalOriginOp<bool>> CreateClearStoragesForOriginPrefixOp(
+RefPtr<ResolvableNormalOriginOp<OriginMetadataArray, true>>
+CreateClearStoragesForOriginPrefixOp(
     MovingNotNull<RefPtr<QuotaManager>> aQuotaManager,
     const Maybe<PersistenceType>& aPersistenceType,
     const mozilla::ipc::PrincipalInfo& aPrincipalInfo);
 
-RefPtr<ResolvableNormalOriginOp<bool>> CreateClearDataOp(
+RefPtr<ResolvableNormalOriginOp<OriginMetadataArray, true>> CreateClearDataOp(
     MovingNotNull<RefPtr<QuotaManager>> aQuotaManager,
     const OriginAttributesPattern& aPattern);
 
-RefPtr<ResolvableNormalOriginOp<bool>> CreateShutdownOriginOp(
-    MovingNotNull<RefPtr<QuotaManager>> aQuotaManager,
-    Maybe<PersistenceType> aPersistenceType,
-    const mozilla::ipc::PrincipalInfo& aPrincipalInfo);
+RefPtr<ResolvableNormalOriginOp<OriginMetadataArray, true>>
+CreateShutdownOriginOp(MovingNotNull<RefPtr<QuotaManager>> aQuotaManager,
+                       Maybe<PersistenceType> aPersistenceType,
+                       const mozilla::ipc::PrincipalInfo& aPrincipalInfo);
 
 RefPtr<ResolvableNormalOriginOp<bool>> CreateShutdownClientOp(
     MovingNotNull<RefPtr<QuotaManager>> aQuotaManager,
